02.calendar/cal.rb

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require 'date'
+require 'optparse'
+
+year = Date.today.year
+month = Date.today.month
+
+opt = OptionParser.new
+opt.on('-y VAL') { |v| year = v.to_i }
+opt.on('-m VAL') { |v| month = v.to_i }
+opt.parse!(ARGV)
+
+selected_date = Date.new(year, month, 1)
+dates = [Array.new(7, 0)]
+date_index = selected_date.cwday % 7
+dates[0][date_index] = 1
+date_index += 1
+
+(2..selected_date.next_month.prev_day.day).each do |date|
+  if date_index == 7
+    dates << Array.new(7, 0)
+    date_index = 0
+  end
+
+  dates[-1][date_index] = date
+  date_index += 1
+end
```

Comment:
datesを生成する処理はこんな感じにするといいかもしれません。

```suggestion
first_date = Date.new(year, month, 1)
last_date = Date.new(year, month, -1)
dates = []
(first_date..last_date).each do |date|
  if date.sunday? || date.day == 1
    dates << []
  end
  dates[-1][date.wday] = date.day
end
```

ただし、「へー」と思ってコピペしておしまい、ではなく、1行ずつ何をどうしているのかじっくりコードを読んで理解してください。
もし難しいところや不明な点があれば気軽に聞いてください！